inginious/backend/backend.py

```diff
@@ -138,17 +138,16 @@ async def handle_client_new_job(self, client_addr, message: ClientNewJob):
         self._logger.info("Adding a new job %s %s to the queue", client_addr, message.job_id)
         job = WaitingJob(message.priority, time.time(), client_addr, message.job_id, message)
         self._waiting_jobs[message.job_id] = job
-        self._waiting_jobs_pq.put((message.environment_type, message.environment, message.environment_parameters["ssh_allowed"]), job)
+        self._waiting_jobs_pq.put((message.environment_type, message.environment, message.environment_parameters["ssh_allowed"]), message.job_id)
```

Comment:
If you only push the job_id, the heap will be ordered by job_id. You should push a tuple (priority, time of arrival in the queue, job_id) or something like that instead.

Reply:
Fixed